src/newrelic/processor.go

```diff
@@ -560,48 +598,53 @@ func harvestPayload(p PayloadCreator, args *harvestArgs) {
 	}
 }
 
-func considerHarvestPayload(p PayloadCreator, args *harvestArgs) {
+func considerHarvestPayload(p PayloadCreator, args *harvestArgs, duc dataUsageController) {
 	if p.Empty() {
 		return
 	}
 
+	duc.wg.Add(1)
 	if args.blocking {
 		// Invoked primarily by CleanExit
-		harvestPayload(p, args)
+		harvestPayload(p, args, duc)
 	} else {
-		go harvestPayload(p, args)
+		go harvestPayload(p, args, duc)
 	}
 }
 
-func considerHarvestPayloadTxnEvents(txnEvents *TxnEvents, args *harvestArgs) {
+func considerHarvestPayloadTxnEvents(txnEvents *TxnEvents, args *harvestArgs, duc dataUsageController) {
 	if args.splitLargePayloads && (txnEvents.events.Len() >= (limits.MaxTxnEvents / 2)) {
 		events1, events2 := txnEvents.Split()
-		considerHarvestPayload(&TxnEvents{events1}, args)
-		considerHarvestPayload(&TxnEvents{events2}, args)
+		considerHarvestPayload(&TxnEvents{events1}, args, duc)
```

Comment:
Have you been able to test if this works if a payload is split?  Maybe test by setting the max payload size small enough to trigger it with a normal payload.

Reply:
Added a test for this in [test(daemon): add usage metrics test for multiple same-endpoint metrics](https://github.com/newrelic/newrelic-php-agent/pull/507/commits/68408bd2fa7a2ecd4df2fd02bc68b9f95f385ac2)